libtiledbvcf/src/c_api/tiledbvcf.cc

```diff
@@ -954,6 +954,50 @@ int32_t tiledb_vcf_reader_set_scan_all_samples(
   return TILEDB_VCF_OK;
 }
 
+int32_t tiledb_vcf_reader_get_stats_buffer_sizes(
+    tiledb_vcf_reader_t* reader, size_t* num_rows, size_t* alleles_size) {
+  if (sanity_check(reader) == TILEDB_VCF_ERR || num_rows == nullptr ||
+      alleles_size == nullptr)
+    return TILEDB_VCF_ERR;
+
+  std::tuple<size_t, size_t> sizes;
+  if (SAVE_ERROR_CATCH(reader, sizes = reader->reader_->stats_buffer_sizes()))
+    return TILEDB_VCF_ERR;
+
+  *num_rows = std::get<0>(sizes);
+  *alleles_size = std::get<1>(sizes);
+
+  return TILEDB_VCF_OK;
+}
+
+int32_t tiledb_vcf_reader_prepare_stats(tiledb_vcf_reader_t* reader) {
```

Comment:
tiledb_vcf_reader_prepare_variant_stats

I think you get the idea. I'll stop flagging those in the review comments.